sky/packages/sky/lib/animation/scroll_behavior.dart

```diff
@@ -22,27 +22,28 @@ class BoundedBehavior extends ScrollBehavior {
 
   double _contentsSize;
   double get contentsSize => _contentsSize;
-  void set contentsSize (double value) {
-    if (_contentsSize != value) {
-      _contentsSize = value;
-      // TODO(ianh) now what? what if we have a simulation ongoing?
-    }
-  }
 
   double _containerSize;
   double get containerSize => _containerSize;
-  void set containerSize (double value) {
-    if (_containerSize != value) {
-      _containerSize = value;
-      // TODO(ianh) now what? what if we have a simulation ongoing?
-    }
+
+  /// Returns the new scrollOffset.
+  double updateSize({
+      double contentsSize,
+      double containerSize,
+      double scrollOffset: 0.0
+  }) {
+    if (contentsSize != null)
+      _contentsSize = contentsSize;
+    if (containerSize != null)
+      _containerSize = containerSize;
+    return scrollOffset.clamp(minScrollOffset, maxScrollOffset);
```

Comment:
Sorry, I was concerned that contentSize or containerSize might be null. They're initialized to 0.